src/mlpack/tests/init_rules_test.cpp

```diff
@@ -126,6 +129,35 @@ BOOST_AUTO_TEST_CASE(OivsInitTest)
   BOOST_REQUIRE_EQUAL(1, 1);
 }
 
+BOOST_AUTO_TEST_CASE(XavierInitTest)
+{
+  arma::mat weights;
+  //RandomSeed(21);
+  XavierInit xavierUniform;
+  XavierInit xavierNormal;
+  xavierUniform.Initialize<XavierUniform>(weights, 100, 100);
```

Comment:
I did try the solution you suggested previously. The problems were if you use something like this 
template<typename InitializerType>
class XavierInit
{
.....
......
 template<typename eT, typename InitType = InitializerType>
  void Initialize(
....
}
}
This would work fine in GCC, but I think it does not work in [vc++](https://stackoverflow.com/questions/6972368/stdenable-if-to-conditionally-compile-a-member-function).  Second ans comments. Hence that's why i choose this approach. This is equivalent to partial specialisation of functions which i think is not permitted in c++11. Correct me if i am wrong.